File: src/target26_1/java/com/skysoft/mixin/ScoreboardHudMixin.java
package com.skysoft.mixin;

import com.llamalad7.mixinextras.injector.wrapmethod.WrapMethod;
import com.llamalad7.mixinextras.injector.wrapoperation.Operation;
import com.skysoft.features.misc.ScoreboardPositionEditor;
import net.minecraft.client.gui.GuiGraphicsExtractor;
import net.minecraft.client.gui.Gui;
import net.minecraft.world.scores.Objective;
import org.spongepowered.asm.mixin.Mixin;

@Mixin(Gui.class)
public abstract class ScoreboardHudMixin {
    @WrapMethod(method = "displayScoreboardSidebar")
    private void skysoftPositionScoreboard(GuiGraphicsExtractor context, Objective objective, Operation<Void> original) {
        ScoreboardPositionEditor.INSTANCE.render(context, objective, () -> { original.call(context, objective); return kotlin.Unit.INSTANCE; });
    }
}

File: src/target26_2/kotlin/com/skysoft/mixin/ScoreboardHudMixin.kt
package com.skysoft.mixin

import com.llamalad7.mixinextras.injector.wrapmethod.WrapMethod
import com.llamalad7.mixinextras.injector.wrapoperation.Operation
import com.skysoft.features.misc.ScoreboardPositionEditor
import net.minecraft.client.gui.GuiGraphicsExtractor
import net.minecraft.client.gui.Hud
import net.minecraft.world.scores.Objective
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.gen.Invoker

@Mixin(Hud::class)
interface ScoreboardHudAccessor {
    @Invoker("displayScoreboardSidebar")
    fun skysoftDisplayScoreboardSidebar(context: GuiGraphicsExtractor, objective: Objective)
}

@Mixin(Hud::class)
abstract class ScoreboardHudMixin {
    @WrapMethod(method = ["displayScoreboardSidebar"])
    private fun skysoftPositionScoreboard(
        context: GuiGraphicsExtractor,
        objective: Objective,
        original: Operation<Void>,
    ) {
        ScoreboardPositionEditor.render(context, objective) {
            original.call(context, objective)
        }
    }
}
